### PR TITLE
Added chrome.i18n.detectLanguage

### PR DESCRIPTION
**Pull request by KeithHenry**
`chrome.i18n.detectLanguage` is missing from the type definitions, this change adds the definition.

Original documentation at https://developer.chrome.com/extensions/i18n#method-detectLanguage

### CODE REVIEW

File: chrome/index.d.ts [hunk 1]
```diff
@@ -3324,6 +3324,12 @@ declare namespace chrome.i18n {
      * @since Chrome 35.
      */
     export function getUILanguage(): string;
+    
+    /** Detects the language of the provided text using CLD.
+    * @param text User input string to be translated.
+    * @param callback The callback parameter should be a function that looks like this: function(object result) {...};
+    */
+    export function detectLanguage(text: string, callback:(result:any) => void): void;
```
Comment (paulvanbrenk) [inline chrome/index.d.ts]: looking at the docs, the callback can be specified more precisely than `any`.

**Comment by KeithHenry** [comment]
Probably, but at the moment it's not specified at all.

**Comment by KeithHenry** [comment]
I'll take a look tomorrow.

**Comment by paulvanbrenk** [comment]
I'm fine with pulling this in as is, but want to wait if someone else wants to chime in.

File: chrome/index.d.ts (hunk 1 above)
Comment (KeithHenry) [inline chrome/index.d.ts]: I've had a look, something like:

```
    /** Holds detected ISO language code and its percentage in the input string */
    interface DetectedLanguage {
        /** An ISO language code such as 'en' or 'fr'. 
         * For a complete list of languages supported by this method, see  [kLanguageInfoTable]{@link https://src.chromium.org/viewvc/chrome/trunk/src/third_party/cld/languages/internal/languages.cc}. 
         * For an unknown language, 'und' will be returned, which means that [percentage] of the text is unknown to CLD */
        language: string;

        /** The percentage of the detected language */
        percentage: number; 
    }

    /** Holds detected language reliability and array of DetectedLanguage */
    interface LanguageDetectionResult {
        /** CLD detected language reliability */
        isReliable: boolean;

        /** array of detectedLanguage */
        languages: DetectedLanguage[];
    }
```
I'll set up another pull

**Comment by andy-ms** [comment]
We've recently updated the repository structure.
All packages are now in a `types` subdirectory.
Please rebase this pull request onto the latest `master` branch.

**Comment by KeithHenry** [comment]
@andy-ms I've made that change in a new PR https://github.com/DefinitelyTyped/DefinitelyTyped/pull/15405

This change already appears to be against `DefinitelyTyped:master`, should I rebase it to a different branch?

**Comment by andy-ms** [comment]
Well, if you've already made a new PR, I'll close this and let that one replace this.

**Pull request closed without merging**